verification/src/http_server/handlers/verification/sourcify/api.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{VerificationResponse, VerificationResult};
+use actix_web::{error, error::Error};
+use reqwest::Url;
+use std::sync::Arc;
+
+use super::types::{ApiFilesResponse, ApiRequest, ApiVerificationResponse, Files};
+
+#[async_trait::async_trait]
+pub(super) trait SourcifyApi {
+    async fn verification_request(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error>;
+
+    async fn source_files_request(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiFilesResponse, reqwest::Error>;
+}
+
+pub struct SourcifyApiClient {
+    host: Url,
+    request_timeout: u64,
+    verification_attempts: u64,
+}
+
+impl SourcifyApiClient {
+    pub fn new(host: Url, request_timeout: u64, verification_attempts: u64) -> Self {
+        Self {
+            host,
+            request_timeout,
+            verification_attempts,
+        }
+    }
+
```

Comment:
Probably forget to remove underscore

Reply:
no, it is like private inner function for `verification_request`, that will be called several times (see below)

Reply:
I agree that naming may be wrong, maybe call it `verification_request_once` ?